src/internal/m365/exchange/events_restore.go

```diff
@@ -210,15 +212,108 @@ func updateExceptionOccurrences(
 
 		evt = toEventSimplified(evt)
 
-		// TODO(meain): Update attachments (might have to diff the
-		// attachments using ids and delete or add). We will have
-		// to get the id of the existing attachments, diff them
-		// with what we need a then create/delete items kinda like
-		// permissions
 		_, err = ac.PatchItem(ctx, userID, ptr.Val(evts[0].GetId()), evt)
 		if err != nil {
 			return clues.Wrap(err, "updating event instance")
 		}
+
+		evt, err = api.EventFromMap(instance)
+		if err != nil {
+			return clues.Wrap(err, "parsing exception event (second)")
+		}
+
+		err = updateAttachments(ctx, ac, userID, containerID, ptr.Val(evts[0].GetId()), evt)
+		if err != nil {
+			return clues.Wrap(err, "updating attachments")
+		}
+	}
+
+	return nil
+}
+
+func updateAttachments(
```

Comment:
I'm a bit concerned with the higher level algorithm in use here.  So far you have an O(2n^2), which is a bit of a smell, though possibly fine if we're assuming a small n and minimal processing for each step.  If we expect larger sizes of n, or extra processing (such as api calls) within any of these steps, you might want to refactor this to a more streamlined process.  Implementation for a refactor like that looks a little a little messy, so I don't want to push this, so long as performance can stay within reason.

Reply:
`n` should be pretty small in most cases as these are attachments to events. Also, the comparison within them will mostly be just comparing two strings which should not be that expensive.
Related: https://github.com/alcionai/corso/pull/3644#discussion_r1235604141